server/game/cards/02_SHD/upgrades/VambraceFlamethrower.ts

```diff
@@ -11,11 +12,14 @@ export default class VambraceFlamethrower extends UpgradeCard {
     }
 
     public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
         this.addGainOnAttackAbilityTargetingAttached({
             title: 'Deal 3 damage divided as you choose among enemy ground units',
             immediateEffect: AbilityHelper.immediateEffects.distributeDamageAmong({
                 amountToDistribute: 3,
-                canChooseNoTargets: true,
+                optional: true,
```

Comment:
Ah sorry, I was unclear - I meant for "optional" to be added at the ability level, not inside of `distributeDamageAmong`. Using optional inside of an `immediateEffect` is not well-supported, I need to decide if we just want to remove it altogether.

I think the change you made to `DistributeAmongTargetsSystem` can still work though, see my comment there